common/stacks/src/main/scala/weco/api/stacks/http/SierraSource.scala

```diff
@@ -31,33 +37,67 @@ class SierraSource(client: HttpClient with HttpGet with HttpPost)(
   private implicit val umItemStub: Unmarshaller[HttpEntity, SierraItemData] =
     CirceMarshalling.fromDecoder[SierraItemData]
 
+  private implicit val umItemEntriesStub
+    : Unmarshaller[HttpEntity, SierraItemDataEntries] =
+    CirceMarshalling.fromDecoder[SierraItemDataEntries]
+
   private implicit val umErrorCode: Unmarshaller[HttpEntity, SierraErrorCode] =
     CirceMarshalling.fromDecoder[SierraErrorCode]
 
+  /** Returns data for a list of items
+    *
+    * Note: if some of the IDs requested do not exist, they will
+    * not appear in the results list. It's up to the consumer to
+    * identify that and react to it as required.
```

Comment:
Should we drop a warning log for this? It's not something that should happen very often in practice.

Reply:
I've reconsidered this and now return an error state reporting on what's missing and what's returned. Realising Sierra would 404 in the case of no matches made me think we should provide a slightly nicer interface, rather than pushing that weirdness up to consumers.